### PR TITLE
172 Product Listing Page

### LINKED ISSUE #205: Product Listing Page
Product page is already implemented, Product listing needs to be populated from Coveo.

For Example:
https://www.abcam.com/en-in/products/primary-antibodies
https://www.abcam.com/en-us/products/secondary-antibodies

### PR DESCRIPTION
**Pull request by Rajesh-GSPANN**
Hi @rgravitvl , Can you please review the PR and Merge

Fix #205 

Test URLs:
- Before: [Secondar Antibodies](https://main--danaher-optimus--aemsites.aem.page/en-us/products/secondary-antibodies)
- After: [Secondary Antibodies](https://172-product-listing-v2--danaher-optimus--aemsites.hlx.page/en-us/products/secondary-antibodies)

### CODE REVIEW

File: blocks/product-list/product-list.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  const productFullResponse = await getProductsListResponse(result);
+  const parsedJson = JSON.parse(productFullResponse);
+  const productsListResponse = parsedJson.results;
+
+  renderRecords(productsListResponse, block);
+  const totalRecords = parsedJson.totalCount;
+  const totalPages = Math.ceil(totalRecords / pageSize);
+
+  const paginationContainer = div({ class: 'pagination-controls flex flex-wrap justify-center items-center gap-2 mt-8' });
+  paginationContainer.innerHTML = '';
```
Comment (pardeepgera23) [inline blocks/product-list/product-list.js:66]: Avoid clearing out the innerHTML of containers

File: blocks/product-list/product-list.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    const productTagsButton = button({ class: 'product-tags appearance-none px-2 py-1 rounded-e text-xs font-semibold tracking-wider break-keep bg-[rgb(237,246,247)] text-[rgb(44,101,107)] border-[rgb(44,101,107)] border' });
+    productTagsButton.appendChild(span({ class: 'pt-0' }, item));
+    productTagsDiv.appendChild(productTagsButton);
+  });
+  return productTagsDiv;
+}
+
+// Function to render records on the page
+function renderRecords(records, block) {
+  block.innerHTML = ''; // Clear previous content
```
Comment (pardeepgera23) [inline blocks/product-list/product-list.js:26]: Avoid clearing out the innerHTML of containers

File: blocks/product-list/product-list.js (hunk 1 above)
Comment (Rajesh-GSPANN) [reply blocks/product-list/product-list.js]: Hi @pardeepgera23 ,
I have made the changes can you please have a look at it.

File: blocks/product-list/product-list.js (hunk 2 above)
Comment (Rajesh-GSPANN) [reply blocks/product-list/product-list.js]: Hi @pardeepgera23 ,
I have made the changes can you please have a look at it.

**Review by pardeepgera23 — APPROVED** [review]

**Comment by Rajesh-GSPANN** [comment]
Hi @rgravitvl ,
Can you please review the PR from your side.

CC: @Meenu-GSPANN @pardeepgera23

**Review by rgravitvl — APPROVED** [review]

**Pull request merged**